debugger/Main.hs

```diff
@@ -21,19 +21,126 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+addOption config (OutputDir d)     = return config { confOutputDir = d }
+addOption config Help              = return config { confAction = ActionHelp}
+addOption config OutputInternal   = return config { confOutputInternal = True }
+addOption config (OutputInput p)  = return config { confOutputInput = p }
+addOption config DumpFlat         = return config { confDumpFlat = True }
+
+validateConfig :: Config -> IO ()
+validateConfig Config{..} = do
+    when (confDatalogFile == "" && confAction /= ActionHelp && confAction /= ActionVersion)
+         $ errorWithoutStackTrace "input file not specified"
 
 main :: IO ()
 main = do
-       (filename:_) <- getArgs
-       contents <- readFile filename
-       let result = parseText contents
-       putStr result
+       args <- getArgs
+       prog <- getProgName
+       home <- lookupEnv "DDLOG_HOME"
+       config <- case getOpt Permute options args of
+                      (flags, [], []) -> do
+                                            conf <- foldM addOption defaultConfig flags
+                                            validateConfig conf
+                                            return conf
+                                         `catch`
+                                         (\e -> do putStrLn $ usageInfo ("Usage: " ++ prog ++ " [OPTION...]") options
+                                                   throw (e::SomeException))
+                      _ -> errorWithoutStackTrace $ usageInfo ("Usage: " ++ prog ++ " [OPTION...]") options
+       config' <- case home of
+           Just(p) -> addOption config (LibDir $ p ++ "/lib")
+           _       -> return config
+       do
+            datalogProg <- parseProgram config'
+            events <- parseEventFromDumpFile config'
+            let recordMap = handleDebugEvents events emptyDebuggerMaps datalogProg
+                s = queryAll events (dbgRecordMap recordMap)
+            putStr $ show s
+
+queryAll :: [Event] -> DebuggerRecordMap -> String
+queryAll [] _ = "Finished"
+queryAll (event: events) dgbRecordMap =
+    let outputRecord = evtOutput event
+        operatorId = evtOperatorId event
+        operatorInput = InputOp operatorId
+        dbgRecord = DebuggerRecord {dbgRecord=outputRecord, dbgOperatorId=operatorInput}
+        dbgRecordNodes = queryDerivations dbgRecord dgbRecordMap
+    in trace (show dbgRecordNodes) $ queryAll events dgbRecordMap
+
+parseProgram :: Config -> IO (DatalogProgram)
+parseProgram Config{..} = do
+    fdata <- readFile confDatalogFile
+    (d, _, _) <- parseDatalogProgram (takeDirectory confDatalogFile:confLibDirs) True fdata confDatalogFile
+    d'' <- case confOutputInternal of
+        False -> return d
+        True ->  return $ progOutputInternalRelations d
+    d''' <- case confOutputInput of
+         "" -> return d''
+         x  ->  return $ progMirrorInputRelations d'' x
+    when confDumpFlat $
```

Comment:
I don't think we need to add this option to dump the transformed program in the debugger program. If we want to dump it, we can always use the ddlog binary to do it.